engine/src/main/java/org/pentaho/di/job/entries/folderscompare/JobEntryFoldersCompare.java

```diff
@@ -0,0 +1,628 @@
+    return je;
+  }
+
+  public String getXML() {
+    StringBuilder retval = new StringBuilder( 200 ); // 133 chars in just spaces and tag names alone
+
+    retval.append( super.getXML() );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "include_subfolders", includesubfolders ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "compare_filecontent", comparefilecontent ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "compare_filesize", comparefilesize ) );
+
+    retval.append( "      " ).append( XMLHandler.addTagValue( "compareonly", compareonly ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "wildcard", wildcard ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "filename1", filename1 ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "filename2", filename2 ) );
+    if ( parentJobMeta != null ) {
+      parentJobMeta.getNamedClusterEmbedManager().registerUrl( filename1 );
+      parentJobMeta.getNamedClusterEmbedManager().registerUrl( filename2 );
+    }
+    return retval.toString();
+  }
+
+  public void loadXML( Node entrynode, List<DatabaseMeta> databases, List<SlaveServer> slaveServers,
+    Repository rep, IMetaStore metaStore ) throws KettleXMLException {
+    try {
+      super.loadXML( entrynode, databases, slaveServers );
+      includesubfolders = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "include_subfolders" ) );
+      comparefilecontent = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "compare_filecontent" ) );
+      comparefilesize = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "compare_filesize" ) );
+
+      compareonly = XMLHandler.getTagValue( entrynode, "compareonly" );
+      wildcard = XMLHandler.getTagValue( entrynode, "wildcard" );
+      filename1 = XMLHandler.getTagValue( entrynode, "filename1" );
+      filename2 = XMLHandler.getTagValue( entrynode, "filename2" );
+    } catch ( KettleXMLException xe ) {
+      throw new KettleXMLException( BaseMessages.getString( PKG, "JobFoldersCompare.Meta.UnableLoadXML", xe
+        .getMessage() ) );
+    }
+  }
+
+  public void loadRep( Repository rep, IMetaStore metaStore, ObjectId id_jobentry, List<DatabaseMeta> databases,
+    List<SlaveServer> slaveServers ) throws KettleException {
+    try {
+      includesubfolders = rep.getJobEntryAttributeBoolean( id_jobentry, "include_subfolders" );
+      comparefilecontent = rep.getJobEntryAttributeBoolean( id_jobentry, "compare_filecontent" );
+      comparefilesize = rep.getJobEntryAttributeBoolean( id_jobentry, "compare_filesize" );
+
+      compareonly = rep.getJobEntryAttributeString( id_jobentry, "compareonly" );
+      wildcard = rep.getJobEntryAttributeString( id_jobentry, "wildcard" );
+      filename1 = rep.getJobEntryAttributeString( id_jobentry, "filename1" );
+      filename2 = rep.getJobEntryAttributeString( id_jobentry, "filename2" );
+    } catch ( KettleException dbe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)